### PR TITLE
ARN object model class

### PR DESCRIPTION
**Pull request by jonathan-gramain**
Will be used by backbeat for roles, and potentially can be used for other components that manipulate ARNs.

### CODE REVIEW

**Comment by jonathan-gramain** [comment]
@ironman-machine try

File: lib/models/ARN.js [hunk 1]
```diff
@@ -0,0 +1,94 @@
+const errors = require('../errors');
+
+const validServices = {
+    aws: ['s3', 'iam'],
```
Comment (rahulreddy) [inline lib/models/ARN.js]: We now have `sso`, `ring`, `utapi`, `sts` as valid services for ARN.
https://github.com/scality/Arsenal/blob/master/lib/policyEvaluator/RequestContext.js#L133
May be we should include them as well.

File: lib/models/ARN.js [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+};
+
+class ARN {
+    constructor(partition, service, region, accountId, resource) {
```
Comment (rahulreddy) [inline lib/models/ARN.js]: Please document the constructor params.

File: lib/models/ARN.js (hunk 1 above)
Comment (jonathan-gramain) [reply lib/models/ARN.js]: updated

**Comment by rahulreddy** [comment]
@ironman-machine r+

**Comment by rahulreddy** [comment]
@ironman-machine r- r+

**Pull request merged**